### ISSUE TITLE
NuGet errors with > 3.28

### ISSUE THREAD

**Issue by UweKeim** [oleg-shilo/cs-script#190]
This line worked for years:

```cs
//css_nuget -force:3600 -ng:"-Source https://nuget.org/api/v2/" MoreLinq
```

After upgrading to CS-Script 3.30, I randomly/often get errors:

> NuGet> Processing NuGet packages...
> NuGet shell command:
> \\nas001\data\development\tools\cs-script\lib\nuget.exe install MoreLinq -Source https://nuget.org/api/v2/ -OutputDirectory "C:\ProgramData\CS-Script\nuget\MoreLinq"
> 
> Feeds used:
>   https://nuget.org/api/v2/
> 
> Installing package 'MoreLinq' to 'C:\ProgramData\CS-Script\nuget\MoreLinq'.
>   CACHE https://www.nuget.org/api/v2/FindPackagesById()?id='MoreLinq'&semVerLevel=2.0.0
> Package "MoreLinq.3.3.2" is already installed.
> Error: Specified file could not be compiled.
> 
> > -----
> A new NuGet package has been installed. If some of it's components are not found you may need to restart the script again.
> > -----
> 
> System.ApplicationException: Cannot process NuGet package 'MoreLinq'
>    at csscript.NuGet.Resolve(String[] packages, Boolean suppressDownloading, String script)
>    at csscript.CSExecutor.AggregateReferencedAssemblies(ScriptParser parser)
>    at csscript.CSExecutor.Compile(String scriptFileName)
>    at csscript.CSExecutor.ExecuteImpl()

I downgraded again to 3.28 and everything works fine again.

**Comment by oleg-shilo** [comment]
Thank you for letting me know. Will look at it. 
As far as I remember there was some change in the package dependency discovery mechanism that had some limitations in3.28. Addressing them could have cause these side effects.

**Comment by oleg-shilo** [comment]
To my pleasant surprise the problem is 100% reproducible in my environment. Every time.  
Meaning we can work with it :)

**Comment by oleg-shilo** [comment]
OK. Got it working.

The problem is caused by the `//css_nuget` directive containing invalid package name. The correct package name is all lowercase: https://www.nuget.org/packages/morelinq/  

And v3.28 had a weaker package probing algorithm so it picked the package even in case of name string case mismatch. Later the name matching condition was made stronger and caused this side effect. Thus if you change the directive as below:
```C#
//css_nuget -force:3600 -ng:"-Source https://nuget.org/api/v2/" morelinq
```
Saying that I found matching with `ignoreCase` being more beneficial. Particularly because I found no strict requirements between package name and `nuget.exe` CLI being case sensitive. So I have changed the implementation to be case insensitive. 

The change will be available in the next release.
